pkg/sidecar/pod.go

```diff
@@ -47,7 +48,15 @@ func add(cfg config.Config, logger logr.Logger, otelcol v1beta1.OpenTelemetryCol
 		container.Env = append(container.Env, attributes...)
 	}
 	pod.Spec.InitContainers = append(pod.Spec.InitContainers, otelcol.Spec.InitContainers...)
-	pod.Spec.Containers = append(pod.Spec.Containers, container)
+
+	if featuregate.EnableNativeSidecarContainers.IsEnabled() {
+		policy := corev1.ContainerRestartPolicyAlways
+		container.RestartPolicy = &policy
+		// TODO(frzifus): Add StartupProbe
```

Comment:
Should we pre-define a startup probe that checks the `${service.telemetry.metrics.address}/metrics` endpoint, expose it in the CRD or do something else?

wdyt @open-telemetry/operator-approvers ?

Reply:
Is there a reason to treat it any differently than we do readiness probes and liveness probes? We can even default to the readiness probe if not set.

Reply:
Nope, but reusing the readiness probe sounds good to me :+1:

Reply:
I don't think there's a good reason... is there much of a difference between these things? Maybe in v1beta1 we should just use the upstream definition?

Reply:
Using the upstream definition would extend ours by the [ProbeHandler](https://github.com/kubernetes/api/blob/5147c1a32f6a0b9b155bb84e59f933e0ff8a3792/core/v1/types.go#L2836).